lib/Sema/TypeCheckDistributed.cpp

```diff
@@ -119,28 +137,51 @@ bool swift::checkDistributedFunction(FuncDecl *func, bool diagnose) {
     auto paramTy = func->mapTypeIntoContext(param->getInterfaceType());
     if (TypeChecker::conformsToProtocol(paramTy, encodableType, module).isInvalid() ||
         TypeChecker::conformsToProtocol(paramTy, decodableType, module).isInvalid()) {
-      if (diagnose)
-        func->diagnose(
+      if (diagnose) {
+        auto diag = func->diagnose(
             diag::distributed_actor_func_param_not_codable,
-            param->getArgumentName().str(),
-            param->getInterfaceType()
-        );
-      // TODO: suggest a fixit to add Codable to the type?
+            param->getArgumentName().str(), param->getInterfaceType(),
+            func->getDescriptiveKind(), "Codable");
+        if (auto paramNominalTy = paramTy->getAnyNominal()) {
+          addCodableFixIt(paramNominalTy, diag);
+        } // else, no nominal type to suggest the fixit for, e.g. a closure
+      }
+      return true;
+    }
+
+    if (param->isInOut()) {
+      param->diagnose(
+          diag::distributed_actor_func_inout,
+          param->getName(),
+          func->getDescriptiveKind(), func->getName()
+      ).fixItRemove(param->getSpecifierLoc()); // FIXME(distributed): does not seem to cause the fixit...? Is the Loc invalid...?
```

Comment:
This is weird... should work, right? But it does not seem to cause the fixit (i verified using the `...}}{{12-14=}}` and nothing 🤔

Reply:
@bnbarham maybe knows?

Reply:
So it turns out this SpecifierLoc is `Invalid` 😢  Not sure why or how we'd fix it...

The func is:

```
(parameter "burger" apiName=inNOut type='String' interface type='String' inout)

  distributed func noInout(inNOut burger: inout String) { }
```

so I don't see what's "weird" about it hm...

Reply:
I found a workaround and asking around for help how to do it properly :)

Reply:
Oh boy, this is a bug with the SpecifierLoc -- it is _never_ correct currently.

I'll file a bug and work on it.

Reply:
https://bugs.swift.org/browse/SR-15486